## Escopo escolhido para o projeto

O projeto utilizará como base principal as tabelas de alfabetização em nível de UF e Brasil.

As tabelas escolhidas para o MVP são:

- `uf`: resultados observados da alfabetização por unidade da federação e rede de ensino;
- `meta_alfabetizacao_uf`: metas de alfabetização por UF até 2030;
- `meta_alfabetizacao_brasil`: metas e resultado consolidado em nível nacional.

Essa escolha permite analisar a situação da alfabetização no Brasil a partir de três pontos: resultado observado, distância em relação às metas e comparação com o cenário nacional.

In [1]:
import pandas as pd
from pathlib import Path

RAW_DIR = Path("../data/raw")

df_uf = pd.read_csv(RAW_DIR / "br_inep_avaliacao_alfabetizacao_uf.csv.gz", compression="gzip")
df_meta_uf = pd.read_csv(RAW_DIR / "br_inep_avaliacao_alfabetizacao_meta_alfabetizacao_uf.csv.gz", compression="gzip")
df_meta_brasil = pd.read_csv(RAW_DIR / "br_inep_avaliacao_alfabetizacao_meta_alfabetizacao_brasil.csv.gz", compression="gzip")

print("uf:", df_uf.shape)
print("meta_uf:", df_meta_uf.shape)
print("meta_brasil:", df_meta_brasil.shape)

uf: (145, 15)
meta_uf: (54, 12)
meta_brasil: (3, 11)


In [2]:
print("Colunas meta_uf:")
print(df_meta_uf.columns.tolist())

print("\nColunas meta_brasil:")
print(df_meta_brasil.columns.tolist())

Colunas meta_uf:
['ano', 'sigla_uf', 'rede', 'taxa_alfabetizacao', 'meta_alfabetizacao_2024', 'meta_alfabetizacao_2025', 'meta_alfabetizacao_2026', 'meta_alfabetizacao_2027', 'meta_alfabetizacao_2028', 'meta_alfabetizacao_2029', 'meta_alfabetizacao_2030', 'percentual_participacao']

Colunas meta_brasil:
['ano', 'rede', 'taxa_alfabetizacao', 'meta_alfabetizacao_2024', 'meta_alfabetizacao_2025', 'meta_alfabetizacao_2026', 'meta_alfabetizacao_2027', 'meta_alfabetizacao_2028', 'meta_alfabetizacao_2029', 'meta_alfabetizacao_2030', 'percentual_participacao']


In [3]:
df_meta_uf.head()

,ano,sigla_uf,rede,taxa_alfabetizacao,meta_alfabetizacao_2024,meta_alfabetizacao_2025,meta_alfabetizacao_2026,meta_alfabetizacao_2027,meta_alfabetizacao_2028,meta_alfabetizacao_2029,meta_alfabetizacao_2030,percentual_participacao
0,2024,RR,Pública,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,2023,RR,Pública,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,2023,AC,Pública,NaN,NaN,56.9,62.2,67.3,72.0,76.2,80.0,NaN
3,2024,AC,Pública,51.38,NaN,56.9,62.2,67.3,72.0,76.2,80.0,80.87
4,2023,DF,Pública,NaN,NaN,63.1,67.0,70.6,74.0,77.1,80.0,NaN


In [4]:
df_meta_brasil.head()

,ano,rede,taxa_alfabetizacao,meta_alfabetizacao_2024,meta_alfabetizacao_2025,meta_alfabetizacao_2026,meta_alfabetizacao_2027,meta_alfabetizacao_2028,meta_alfabetizacao_2029,meta_alfabetizacao_2030,percentual_participacao
0,2025,Pública,66.0,60.0,64.00,67.00,71.00,74.00,77.00,80,88.00
1,2024,Pública,59.2,59.9,63.77,67.47,70.97,74.23,77.24,80,87.37
2,2023,Pública,55.9,59.9,63.77,67.47,70.97,74.23,77.24,80,86.00


In [5]:
df_meta_uf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ano                      54 non-null     int64  
 1   sigla_uf                 54 non-null     object 
 2   rede                     54 non-null     object 
 3   taxa_alfabetizacao       50 non-null     float64
 4   meta_alfabetizacao_2024  48 non-null     float64
 5   meta_alfabetizacao_2025  52 non-null     float64
 6   meta_alfabetizacao_2026  52 non-null     float64
 7   meta_alfabetizacao_2027  52 non-null     float64
 8   meta_alfabetizacao_2028  52 non-null     float64
 9   meta_alfabetizacao_2029  52 non-null     float64
 10  meta_alfabetizacao_2030  52 non-null     float64
 11  percentual_participacao  50 non-null     float64
dtypes: float64(9), int64(1), object(2)
memory usage: 5.2+ KB


In [6]:
df_meta_brasil.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ano                      3 non-null      int64  
 1   rede                     3 non-null      object 
 2   taxa_alfabetizacao       3 non-null      float64
 3   meta_alfabetizacao_2024  3 non-null      float64
 4   meta_alfabetizacao_2025  3 non-null      float64
 5   meta_alfabetizacao_2026  3 non-null      float64
 6   meta_alfabetizacao_2027  3 non-null      float64
 7   meta_alfabetizacao_2028  3 non-null      float64
 8   meta_alfabetizacao_2029  3 non-null      float64
 9   meta_alfabetizacao_2030  3 non-null      int64  
 10  percentual_participacao  3 non-null      float64
dtypes: float64(8), int64(2), object(1)
memory usage: 392.0+ bytes


In [7]:
df_meta_uf[df_meta_uf.isna().any(axis=1)].sort_values(["ano", "sigla_uf"])

,ano,sigla_uf,rede,taxa_alfabetizacao,meta_alfabetizacao_2024,meta_alfabetizacao_2025,meta_alfabetizacao_2026,meta_alfabetizacao_2027,meta_alfabetizacao_2028,meta_alfabetizacao_2029,meta_alfabetizacao_2030,percentual_participacao
2,2023,AC,Pública,NaN,NaN,56.9,62.2,67.3,72.0,76.2,80.0,NaN
4,2023,DF,Pública,NaN,NaN,63.1,67.0,70.6,74.0,77.1,80.0,NaN
1,2023,RR,Pública,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,2024,AC,Pública,51.38,NaN,56.9,62.2,67.3,72.0,76.2,80.0,80.87
5,2024,DF,Pública,59.13,NaN,63.1,67.0,70.6,74.0,77.1,80.0,79.83
0,2024,RR,Pública,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [8]:
print("Duplicados uf:")
print(df_uf.duplicated(subset=["ano", "sigla_uf", "serie", "rede"]).sum())

print("\nDuplicados meta_uf:")
print(df_meta_uf.duplicated(subset=["ano", "sigla_uf", "rede"]).sum())

print("\nDuplicados meta_brasil:")
print(df_meta_brasil.duplicated(subset=["ano", "rede"]).sum())

Duplicados uf:
0

Duplicados meta_uf:
0

Duplicados meta_brasil:
0


## Validações iniciais

- `uf`: 145 registros e 15 colunas;
- `meta_alfabetizacao_uf`: 54 registros e 12 colunas;
- `meta_alfabetizacao_brasil`: 3 registros e 11 colunas.

Foram verificadas duplicidades nas chaves principais de cada tabela:

- `uf`: `ano`, `sigla_uf`, `serie`, `rede`;
- `meta_alfabetizacao_uf`: `ano`, `sigla_uf`, `rede`;
- `meta_alfabetizacao_brasil`: `ano`, `rede`.

Nenhuma duplicidade foi encontrada nessas combinações.

Também foram identificados valores nulos na tabela `meta_alfabetizacao_uf`, principalmente em alguns registros de AC, DF e RR.
Esses casos vão ser mantidos na camada Bronze e tratados como alertas de qualidade na camada Silver, sem exclusão automática dos dados.